include/ginkgo/core/solver/batch_tridiagonal_solver.hpp

```diff
@@ -0,0 +1,278 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2023, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_PUBLIC_CORE_SOLVER_BATCH_TRIDIAGONAL_SOLVER_HPP_
+#define GKO_PUBLIC_CORE_SOLVER_BATCH_TRIDIAGONAL_SOLVER_HPP_
+
+
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/batch_diagonal.hpp>
+#include <ginkgo/core/matrix/batch_identity.hpp>
+#include <ginkgo/core/matrix/batch_tridiagonal.hpp>
+#include <ginkgo/core/matrix/identity.hpp>
+
+namespace gko {
+namespace solver {
+
+/**
+ * The approach to be used to solve the batched trdiagonal system on CUDA/HIP
+ * executor.
+ *
+ * Both "WM_pGE_app1" and "WM_pGE_app2" are based on the Wang and Mou parallel
+ * Gaussian Elimination algorithm. The only difference between the two lies is
+ * the parallel scheme; in "WM_pGE_app1" each thread of the subwarp operates on
+ * a single row of the matrix tile, while in "WM_pGE_2", each thread of the
+ * subwarp handles two adjacent rows of the matrix tile.
+ *
+ * The approach- "vendor_provided" makes use of the "gtsv2StridedBatched"
+ * function provided by cuSparse/hipSparse.
+ *
+ */
+enum class batch_tridiag_solve_approach {
+    WM_pGE_app1,
```

Comment:
I think we dont need to so specific here to have varied capitalization ? Can we just use `wm_pge_app1` and `wm_pge_app2`  ? Or maybe something a bit more descriptive would be better `wm_pge_1row` and `wm_pge_2row`  ?